packages/eyes/src/generate-snapshots/generate-snapshots.ts

```diff
@@ -0,0 +1,107 @@
+const StylableWebpackPlugin = require('@stylable/webpack-plugin');
+import path from 'path';
+import webpack from 'webpack';
+import glob from 'glob';
+import {HTMLSnapshotPlugin} from '@stylable/webpack-extensions';
+import {createElement} from 'react';
+import {renderToStaticMarkup} from 'react-dom/server';
+import {generateFilteringLogic} from './filter-logic';
+import {mapSylesToComponents} from './map-styles-to-components';
+import Registry, {importMeta, getCompName, IComponentMetadata} from '@ui-autotools/registry';
+import {consoleLog} from '@ui-autotools/utils';
+import {dedent} from './dedent';
+import {parseSnapshotFilename} from './filename-utils';
+
+const fileNameRegex = /(?:.\/.autotools\/)(.+)/; // Match the file name
+
+async function buildSingleFile(file: string, directory: string, filteringLogic: (stylableModule: any) => any, config: any) {
+  const entryName = file.match(fileNameRegex)![1];
+
+  function render(compiledFile: any, sourceFile: any) {
```

Comment:
Is there a reason why this function needs to be defined inside of `buildSingleFile` instead of the top level?

Reply:
Ah yeah, it needs to have `entryName` in its context